step-automation-packages/step-automation-packages-yaml/src/main/java/step/automation/packages/yaml/deserialization/YamlAlertingRuleActionDeserializer.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.automation.packages.yaml.deserialization;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import step.automation.packages.AutomationPackageNamedEntityUtils;
+import step.core.yaml.deserializers.NamedEntityYamlDeserializer;
+import step.core.yaml.deserializers.StepYamlDeserializer;
+import step.core.yaml.deserializers.StepYamlDeserializerAddOn;
+import step.core.yaml.deserializers.YamlFieldDeserializationProcessor;
+import step.plugins.alerting.rule.condition.AlertingRuleAction;
+
+import java.io.IOException;
+import java.util.List;
+
+@StepYamlDeserializerAddOn(targetClasses = {AlertingRuleAction.class})
+public class YamlAlertingRuleActionDeserializer extends StepYamlDeserializer<AlertingRuleAction> {
```

Comment:
Can't we achieve this with jackson annotations?

Reply:
@jeromecomte 
I agree that jackson annotations would be preferrable, but I wasn't able to perform this deserialization with pure jackson mostly because of existing `@JsonTypeInfo` annotations over target classes.
For yaml format we usually want to specify the type on the top level (to support auto-suggestions via json schema):

```
MyClass:
   fieldA: valueA
   fieldB: valueB
```
But I did't fild a way to apply this format with only jackson annotations - jackson forces to use the `class` field from `@JsonTypeInfo` even if I configure the yaml object mapper with another deserilizer.
Maybe it is possible, but I had not enough time to ivestigate that deeply.